### ISSUE TITLE
Vagrant Consul demo only works with one provider

### ISSUE THREAD

**Issue by dnblankedelman** [hashicorp/consul#2153]
The Vagrant [demo](https://github.com/hashicorp/consul/tree/master/demo/vagrant-cluster) currently only works with the virtualbox provider.  Clearly this wasn't always the case because the README for the demo says:

```
$ vagrant status
Current machine states:
n1                        running (vmware_fusion)
n2                        running (vmware_fusion)
```

It uses debian/wheezy64 which throws the following error if you do indeed try to use the vmware_fusion provider:

> The box you're attempting to add doesn't support the provider
> you requested. Please find an alternate box or use an alternate
> provider. Double-check your requested provider to verify you didn't
> simply misspell it.
> 
> If you're adding a box from HashiCorp's Atlas, make sure the box is
> released.
> 
> Name: debian/wheezy64
> Address: https://atlas.hashicorp.com/debian/wheezy64
> Requested provider: ["vmware_desktop", "vmware_fusion", "vmware_workstation"]

Any chance we could switch that to a box that does support more providers? Thanks!

**Comment by slackpad** [comment]
Hi @dnblankedelman things should work fine with a newer box such as Ubuntu. Would you like to PR this? Thanks!

**Comment by brianshumate** [comment]
Hi @dnblankedelman,

I made some improvements to the demo's `Vagrantfile` in #2523 and you can now specify any Vagrant box you'd prefer to use so that you can specify a box that supports your preferred provider.

With my improvement, you can do the following:

```
DEMO_BOX_NAME="ubuntu/xenial64" vagrant up
```

and the demo will use Ubuntu 16.04 (Xenial) instead of Debian. This is all documented as part of the [README](https://github.com/hashicorp/consul/blob/master/demo/vagrant-cluster/README.md) now, too.

That said, the official Ubuntu Vagrant boxes are built only with VirtualBox support as well. So, provided you have a box in mind that you know is built for the VMware provider, you should be able to use the same syntax as above to specify that box name, and use the demo with VMware instead.

Hope this helps!

**Comment by slackpad** [comment]
Closed via #2523.

**Comment by brianshumate** [comment]
@dnblankedelman No problem! I think the [Bento boxes](https://github.com/chef/bento) have both VirtualBox and Vagrant provider support and their Ubuntu boxes are generally more compatible with Vagrant than are the ones under the *ubuntu* namespace, just as an aside.
